sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -798,7 +804,23 @@ public FileStore getFileStore(Path path) throws IOException {
      */
     @Override
     public void checkAccess(Path path, AccessMode... accessModes) throws IOException {
-        throw new UnsupportedOperationException();
+        if (accessModes != null && accessModes.length != 0) {
+            throw logger.logThrowableAsError(
+                new AccessDeniedException("The access cannot be determined."));
+        }
+        AzurePath.ensureFileSystemOpen(path);
+
+        // Read attributes already wraps BlobStorageException in an IOException.
+        try {
+            readAttributes(path, BasicFileAttributes.class);
+        } catch(IOException e) {
```

Comment:
Just a thought - should auth errors be wrapped into AccessDeniedExceptions?

Reply:
Good question. I tend to think that debugging will be easier if issues with underlying blob stuff is presented differently from nio stuff. I can see an argument where that kind of breaks the point of mapping one to the other, but an AccessDeniedException seems like it would have a response of "well let me go grant access" which is impossible in this case, and an IOException caused by a BlobStorageException kind of leads me down a path of something is wrong with my account or nio configs. What do you think?

Reply:
If that's the general pattern we've aimed for with nio then I'm fine with that.

Reply:
I'm not sure if it's generalized yet haha. Still could change if you think otherwise. But we'd have to add that explicit check in a lot of different places, which I'm also not a fan of because I think we're likely to forget one.